### PR TITLE
Update `rt.conf` to compile all suites with Intel in debug mode

### LINKED ISSUE #898: Compiling all CCPP suites fails with the Intel compiler, always successful with GNU
## Description

Compiling all (or a certain number of) CCPP suites with Intel into the executable causes the compilation to about with an internal compiler error. For each version of the code, this happens above a certain number of CCPP suites, therefore the need to compile multiple versions of the code for regression testing. This also happens on all machines with different versions of the Intel compiler, from 18.0.5 (Hera, Jet, ...) to 2021.2.0 (Cheyenne).

Notes:
- This happens with both DEBUG and PROD flags.
- I tested the option `-qoverride-limits`, this did not help.
- I could **not** reproduce the problem on Intel's development cluster Endeavor using the latest version Intel 2021.4.0. I don't know if this is because of the compiler version or the machine.
- The problem does not and has never occurred with GNU.

### To Reproduce:
- clone the latest developmental code on any tier 1 system
- compile
```
./compile.sh machine.intel '-DAPP=ATM -DDEBUG=[ON|OFF]'
```

### PR DESCRIPTION
**Pull request by climbfuji**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Description

Change the debug compile command for Intel to compile all available suites. This seems to be possible with newer versions of the Intel compilers (it was always possible with GNU). Keep old split compile jobs on outgoing wcoss_dell_p3 and wcoss_cray (there it doesn't work).

Changes are in `tests/rt.conf` only, thus does not affect GNU tests at all.

### Issue(s) addressed

- fixes https://github.com/ufs-community/ufs-weather-model/issues/898

## Testing

Regression tests were run on all platforms that use the Intel compiler (since there are no changes that could possibly affect the GNU runs):

- [x] hera.intel
- (n.n.) hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- (n.n.) cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- (more or less) wcoss_dell_p3*
- (n.n.) opnReqTest for newly added/changed feature
- (n.n.) CI

* on wcoss_dell_p3, the following two tests failed, and I believe for unrelated reasons (at the moment, I can't log into wcoss, for whatever reason, it worked this morning):
```
control_debug
hafs_regional_atm
```
I forgot to use the new `-d` (? ; delete successful rundirs immediately) option, thus running out of disk space might have been the problem.

## Dependencies

None

### CODE REVIEW

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: RT
[RT] Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/944722856/20220523173016/ufs-weather-model
Please make changes and add the following label back: jet-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: RT
[RT] Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/944722856/20220524130007/ufs-weather-model
[RT] Error: Test cpld_restart_c384_p8 011 failed in run_test failed 
[RT] Error: Test hafs_global_multiple_4nests_atm 102 failed in run_test failed 
Please make changes and add the following label back: gaea-intel-RT

**Comment by climbfuji** [comment]
> _**Automated RT Failure Notification**_
> Machine: gaea
> Compiler: intel
> Job: RT
> [RT] Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/944722856/20220524130007/ufs-weather-model
> [RT] Error: Test cpld_restart_c384_p8 011 failed in run_test failed
> [RT] Error: Test hafs_global_multiple_4nests_atm 102 failed in run_test failed
> Please make changes and add the following label back: gaea-intel-RT

@junwang-noaa Looks like gaea tests failed again, but it doesn't seem to be a problem with compiling all suites in debug mode. I'll confirm manually tonight, and will also try on wcoss. My wcoss2 access is not yet working, so I will need someone to help testing.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: RT
[RT] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/944722856/20220605034509/ufs-weather-model
[RT] Error: Test control_wam_debug 093 failed in check_result failed 
[RT] Error: Test control_wam_debug 093 failed in run_test failed 
Please make changes and add the following label back: hera-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: RT
[RT] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/944722856/20220604224515/ufs-weather-model
[RT] Error: Test control_wam_debug 093 failed in check_result failed 
[RT] Error: Test control_wam_debug 093 failed in run_test failed 
Please make changes and add the following label back: orion-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: RT
[RT] Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/944722856/20220605034506/ufs-weather-model
[RT] Error: Test control_wam_debug 092 failed in check_result failed 
[RT] Error: Test control_wam_debug 092 failed in run_test failed 
Please make changes and add the following label back: gaea-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: RT
[RT] Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/944722856/20220605034517/ufs-weather-model
[RT] Error: Test control_c384 018 failed in run_test failed 
[RT] Error: Test control_wam_debug 087 failed in check_result failed 
[RT] Error: Test control_wam_debug 087 failed in run_test failed 
Please make changes and add the following label back: jet-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: RT
[RT] Repo location: /glade/scratch/epicufsrt/autort/tests/auto/pr/944722856/20220604214512/ufs-weather-model
[RT] Error: Test control_wam_debug 093 failed in check_result failed 
[RT] Error: Test control_wam_debug 093 failed in run_test failed 
Please make changes and add the following label back: cheyenne-intel-RT

**Comment by climbfuji** [comment]
@junwang-noaa Also here, please add other reviewers as appropriate. Thanks!

File: tests/rt.conf [hunk 1]
```diff
@@ -120,7 +122,7 @@ RUN     | control_thompson_progcld_thompson_debug
 
 RUN     | regional_debug                                                                                                          |                                         | fv3 |
 
-COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake -D32BIT=ON             |                                         | fv3 |
+COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake -D32BIT=ON             | + wcoss_dell_p3 wcoss_cray              | fv3 |
```
Comment (junwang-noaa) [inline tests/rt.conf:129]: Why do we add dell/cray in the platform list? Does it mean this compile job only works on dell/cray?

File: tests/rt.conf [hunk 2]
```diff
@@ -131,7 +133,7 @@ RUN     | rap_lndp_debug
 RUN     | rap_flake_debug                                                                                                         |                                         | fv3 |
 RUN     | rap_progcld_thompson_debug                                                                                              |                                         | fv3 |
 
-COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP_noah,FV3_RAP_RRTMGP,FV3_RAP_sfcdiff,FV3_RAP_noah_sfcdiff_cires_ugwp,FV3_RRFS_v1beta -D32BIT=ON |                       | fv3 |
+COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP_noah,FV3_RAP_RRTMGP,FV3_RAP_sfcdiff,FV3_RAP_noah_sfcdiff_cires_ugwp,FV3_RRFS_v1beta -D32BIT=ON | + wcoss_dell_p3 wcoss_cray | fv3 |
```
Comment (junwang-noaa) [inline tests/rt.conf:140]: Same here

File: tests/rt.conf (hunk 1 above)
Comment (BrianCurtis-NOAA) [reply tests/rt.conf:129]: With WCOSS dell/cray going away soon, maybe we should remove this?

**Comment by climbfuji** [comment]
@junwang-noaa @BrianCurtis-NOAA Please see the description in the PR above. Compiling all suites doesn't work on the two WCOSS platforms, presumably because of the older Intel compiler. It works everywhere else. Therefore remove them from the new compile line that compiles all suites into one exectuable (`- wcoss_dell_p3 wcoss_cray`) and add them explicitly to the old compile lines (`+ wcoss_dell_p3 wcoss_cray`) - which means that those lines will be ignored for other platforms. This is the syntax we agreed upon a year or so ago in the UFS code manager meetings and that Dusan implemented. Hope this helps.

**Comment by climbfuji** [comment]
@DusanJovic-NOAA Note. I didn't try this on wcoss2 (still haven't had the time to try the tectia client).

**Comment by BrianCurtis-NOAA** [comment]
> @junwang-noaa @BrianCurtis-NOAA Please see the description in the PR above. Compiling all suites doesn't work on the two WCOSS platforms, presumably because of the older Intel compiler. It works everywhere else. Therefore remove them from the new compile line that compiles all suites into one exectuable (`- wcoss_dell_p3 wcoss_cray`) and add them explicitly to the old compile lines (`+ wcoss_dell_p3 wcoss_cray`) - which means that those lines will be ignored for other platforms. This is the syntax we agreed upon a year or so ago in the UFS code manager meetings and that Dusan implemented. Hope this helps.

@junwang-noaa I'm not sure how far in advance of WCOSS 1 shutdown we can plan, but would it make sense to combine all of the debug tests (minus the one that uses MULTI_GASES) and just skip them on dell/cray? WCOSS2 uses the newer intel compiler I believe, so it should work OK there.

**Comment by DusanJovic-NOAA** [comment]
> @DusanJovic-NOAA Note. I didn't try this on wcoss2 (still haven't had the time to try the tectia client).

I will run rt on wcoss2, but today it will be down for maintenance.

**Comment by climbfuji** [comment]
> > @DusanJovic-NOAA Note. I didn't try this on wcoss2 (still haven't had the time to try the tectia client).
> 
> I will run rt on wcoss2, but today it will be down for maintenance.

Thanks, Dusan!

**Comment by climbfuji** [comment]
> > @junwang-noaa @BrianCurtis-NOAA Please see the description in the PR above. Compiling all suites doesn't work on the two WCOSS platforms, presumably because of the older Intel compiler. It works everywhere else. Therefore remove them from the new compile line that compiles all suites into one exectuable (`- wcoss_dell_p3 wcoss_cray`) and add them explicitly to the old compile lines (`+ wcoss_dell_p3 wcoss_cray`) - which means that those lines will be ignored for other platforms. This is the syntax we agreed upon a year or so ago in the UFS code manager meetings and that Dusan implemented. Hope this helps.
> 
> @junwang-noaa I'm not sure how far in advance of WCOSS 1 shutdown we can plan, but would it make sense to combine all of the debug tests (minus the one that uses MULTI_GASES) and just skip them on dell/cray? WCOSS2 uses the newer intel compiler I believe, so it should work OK there.

@BrianCurtis-NOAA I'll leave that up to you and @junwang-noaa to decide!

**Comment by jkbk2004** [comment]
combined and merged with #1256

**Pull request closed without merging**